nncf/onnx/graph/node_utils.py

```diff
@@ -171,10 +171,15 @@ def _get_activation_tensor_shape(
     :param target_point: Determines from input or ouput of a node take a shape info.
     :return: None, if there is no shape info, otherwise - tensor shape.
     """
+    # NOTE: Assumes that all output/input edges for the `node` with `output_port_id`/`input_port_id`
+    # equal to `target_point.port_id` should have the same `tensor_shape` value.
+
     if target_point.type == TargetType.PRE_LAYER_OPERATION:
-        shape = nncf_graph.get_input_edges(node)[target_point.port_id].tensor_shape
```

Comment:
Just to note. There are several places in the NNCF using this approach.

Reply:
Yes, I've changed probably all of them now.